code/game/objects/items/cosmetics.dm

```diff
@@ -305,3 +305,139 @@
 	to_chat(user, "<span class='notice'>You look into the mirror</span>")
 	sleep(150)
 	REMOVE_TRAIT(user, TRAIT_SELF_AWARE, "mirror_trait")
+
+/obj/item/hair_painter
+	name = "hair painter"
+	desc = "A modified airlock painter. Used for changing one's hair colour and occasionally violating safety regulations."
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "hair sprayer"
+	item_state = "hair sprayer"
+	w_class = WEIGHT_CLASS_SMALL
+	flags_1 = CONDUCT_1
+	materials = list(/datum/material/iron=50, /datum/material/glass=50)
+	var/max_dye = 20
+	var/dye_usage = 1
+	var/dye_color = "#ffffff"
+
+/obj/item/hair_painter/Initialize()
+	. = ..()
+	create_reagents(max_dye, REFILLABLE | DRAWABLE) // Easy to pour into, harder to get the reagents out
+
+/obj/item/hair_painter/proc/get_dye()
+	return reagents.get_reagent_amount(/datum/reagent/hair_dye)
+
+/obj/item/hair_painter/proc/use_dye()
+	reagents.remove_reagent(/datum/reagent/hair_dye,dye_usage)
+
+/obj/item/hair_painter/proc/can_use()
+	if(get_dye() >= dye_usage)
+		return TRUE
+	return FALSE
+
+/obj/item/hair_painter/examine(mob/user)
+	. = ..()
+	. += "[src]'s dye storage capacity is [max_dye] unit\s."
+	. += "The gauge reads [reagents.total_volume] unit\s of dye."
+	if (get_dye() != reagents.total_volume)
+		. += "The dye contamination light is lit!"
+
+/obj/item/hair_painter/attack_self(mob/user)
+	dye_color = input(usr,"Choose The Dye Color","Dye Color",dye_color) as color|null
+
+/obj/item/hair_painter/attack(mob/M, mob/user)
+	if(M && ishuman(M) && user.a_intent != INTENT_HARM)
+		if(!can_use())
+			to_chat(user, "<span class='warning'>[src]'s dye tank has no quantum hair dye in it!</span>")
+			return
+		var/mob/living/carbon/human/H = M
+		var/location = user.zone_selected
+		var/mirror = FALSE
+		if(HAS_TRAIT(H, TRAIT_SELF_AWARE) || locate(/obj/structure/mirror) in range(1, H))
```

Comment:
hand mirror

Reply:
Hand mirrors work by adding TRAIT_SELF_AWARE for 15 seconds after using, so it works fine with them